agent/agent.go

```diff
@@ -252,8 +256,7 @@ func (a *Agent) prepareNextTask(ctx context.Context, nextTask *apimodels.NextTas
 func nextTaskHasDifferentTaskGroupOrBuild(nextTask *apimodels.NextTaskResponse, tc *taskContext) bool {
 	if tc.taskConfig == nil ||
 		nextTask.TaskGroup == "" ||
-		nextTask.TaskGroup != tc.taskGroup ||
-		// TODO The Version case is redundant, and can be removed after agents roll over after a deploy.
+		nextTask.TaskGroup != tc.taskGroup || // do we need this case anymore ? I wouldn't think so.
```

Comment:
Hm

Reply:
I'm removing that case and the case below given your TODO assuming that case is actually redundant. Should I be putting a similar TODO above `nextTask.TaskGroup != tc.taskGroup` rather than removing  to handle agent deploys?

Reply:
Yes.